File: apps/admin-console/src/types/ledger.ts
export interface PnLSummaryDTO {
  revenue: number;
  cogs: number;
  gross_profit: number;
  operating_expense: number;
  operating_income: number;
}

export interface IncomeStatementSectionDTO {
  section: string;
  amount: number;
}

export interface IncomeStatementResponseDTO {
  summary: PnLSummaryDTO;
  sections: IncomeStatementSectionDTO[];
  revenue_by_channel: Record<string, number>;
  available_months: string[];
}

File: backend/app/schemas/ledger.py
from typing import List, Dict, Optional
from pydantic import BaseModel

class PnLSummaryDTO(BaseModel):
    revenue: float
    cogs: float
    gross_profit: float
    operating_expense: float
    operating_income: float

class IncomeStatementSectionDTO(BaseModel):
    section: str
    amount: float

class IncomeStatementResponseDTO(BaseModel):
    summary: PnLSummaryDTO
    sections: List[IncomeStatementSectionDTO]
    revenue_by_channel: Dict[str, float]
    available_months: List[str]
